examples/plot_predictive_standard_deviation.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+r"""
+==============================================================
+Comparison of predictive distributions of different regressors
+==============================================================
+
+A simple one-dimensional, noisy regression problem adressed by three different
+regressors:
+
+1. A Gaussian Process
+2. A Random Forest
+3. A Bagging-based Regressor
+
+The regressors are fitted based on noisy observations where the magnitude of
+the noise at the different training point is constant and known. Plotted are
+both the mean and the pointwise 95% confidence interval of the predictions.
+The mean predictions are evaluated on noise-less test data using the mean-
+squared-error. The mean log probabilities of the noise-less test data are used
+to evaluate the predictive distributions (a normal distribution with the
+predicted mean and standard deviation) of the three regressors.
+
+This example is based on the example gaussian_process/plot_gp_regression.py.
+"""
+print(__doc__)
+
+# Author: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+# Licence: BSD 3 clause
+
+import numpy as np
+from scipy.stats import norm
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import RBF
+from sklearn.ensemble import RandomForestRegressor, BaggingRegressor
+from sklearn.metrics import mean_squared_error
+from matplotlib import pyplot as pl
+
+np.random.seed(1)
+
+
+def f(x):
+    """The function to predict."""
+    return x * np.sin(x)
+
+X = np.linspace(0.1, 9.9, 20)
+X = np.atleast_2d(X).T
+
+# Observations and noise
+y = f(X).ravel()
+dy = np.ones_like(y)
+noise = np.random.normal(0, dy)
+y += noise
+
+# Mesh the input space for evaluations of the real function, the prediction and
+# its standard deviation
+x = np.atleast_2d(np.linspace(0, 10, 1000)).T
+
+regrs = {
+    "Gaussian Process": GaussianProcessRegressor(alpha=(dy / y) ** 2,
+                                                 kernel=RBF(1.0),
+                                                 n_restarts_optimizer=100),
```

Comment:
@jmetzen Am I understanding correctly that for `kernel=None`, parameters are fixed and therefore are not tuned? By contrast, if I set `kernel=RBF(1.0)` in combination with `n_restarts_optimizer > 0`, then kernel parameters are optimized? (here the length scale). If this is the case, then maybe we should trigger a warning if `n_restarts_optimizer > 0` and all parameters are `"fixed"`, since there is nothing to optimize.

Also, using these settings, conclusions are different, the mean squared error of GP is worse than the one of bagging...